modules/client-common/src/main/java/org/apache/ignite/internal/client/proto/ClientMessagePacker.java

```diff
@@ -325,6 +326,33 @@ public ClientMessagePacker packUuid(UUID val) {
         return this;
     }
 
+    /**
+     * Writes an {@link IgniteUuid}.
+     *
+     * @param val {@link IgniteUuid} value.
+     * @return This instance.
+     */
+    public ClientMessagePacker packIgniteUuid(IgniteUuid val) {
+        assert !closed : "Packer is closed";
+
+        packExtensionTypeHeader(ClientMsgPackType.IGNITE_UUID, 24);
+
+        // TODO: Pack directly to ByteBuf without allocating IGNITE-15234.
+        var bytes = new byte[24];
+        ByteBuffer bb = ByteBuffer.wrap(bytes);
+
+        UUID globalId = val.globalId();
+
+        bb.putLong(globalId.getMostSignificantBits());
+        bb.putLong(globalId.getLeastSignificantBits());
+
+        bb.putLong(val.localId());
+
+        writePayload(bytes);
+
+        return this;
```

Comment:
```suggestion
        ByteBuffer bb = ByteBuffer.allocate(24);

        UUID globalId = val.globalId();

        bb.putLong(globalId.getMostSignificantBits());
        bb.putLong(globalId.getLeastSignificantBits());

        bb.putLong(val.localId());

        bb.flip();

        writePayload(bb.array());
```
why do we need byte[] wrapping ? = additional inner object creation

Reply:
Well, I still think it should be a part of some other ticket, cause we have same similar logic for packing other data types. If it should be refactored it should be refactored for all cases, WDYT?